### ISSUE TITLE
'newline-after-import' not applied to jsdoc

### ISSUE THREAD

**Issue by emmbm** [import-js/eslint-plugin-import#3125]
The `newline-after-import` rule does not appear to work when the first content encountered after imports is JSDoc. Take the following `eslint.config.js` as an example:

```js
import js from '@eslint/js';
import eslintConfigPrettier from 'eslint-config-prettier';
import eslintPluginImport from 'eslint-plugin-import';
import globals from 'globals';
import ts from 'typescript-eslint';
/**
 * @type {import('eslint').Linter.Config[]}
 */
export default ts.config([
  eslintConfigPrettier,
  js.configs.recommended,
  ...ts.configs.strict,
  ...ts.configs.stylistic,
  {
    plugins: {
      import: eslintPluginImport,
    },
    languageOptions: {
      globals: {
        ...globals.node,
        ...globals.browser,
      },
    },
    rules: {
      'import/newline-after-import': ['error', { count: 1 }],
    },
  },
]);
```

ESLint doesn't indicate the expected error after the import statements, but removing the JSDoc will result in the correct behavior:

```js
import js from '@eslint/js';
import eslintConfigPrettier from 'eslint-config-prettier';
import eslintPluginImport from 'eslint-plugin-import';
import globals from 'globals';
import ts from 'typescript-eslint'; // Expected 1 empty line after import statement not followed by another import.eslint[import/newline-after-import](https://github.com/import-js/eslint-plugin-import/blob/v2.31.0/docs/rules/newline-after-import.md)
export default ts.config([
  eslintConfigPrettier,
  js.configs.recommended,
  ...ts.configs.strict,
  ...ts.configs.stylistic,
  {
    plugins: {
      import: eslintPluginImport,
    },
    languageOptions: {
      globals: {
        ...globals.node,
        ...globals.browser,
      },
    },
    rules: {
      'import/newline-after-import': ['error', { count: 1 }],
    },
  },
]);
```

**Comment by ljharb** [comment]
It's not questionable at all; it would have been a breaking change to default that behavior to true.

**Comment by emmbm** [comment]
Ah, yeah my wording might've come off a bit harsh, didn't mean to! Genuinely curious though now that you mention it, what specific to comments but not code would've made this a breaking change?

**Comment by ljharb** [comment]
It could have created warnings or diffs on codebases that had no warnings or autofix diffs previously.
